WordPress/Classes/Models/WPAccount+RestApi.swift

```diff
@@ -11,49 +11,54 @@ extension WPAccount {
     /// This was done in the context of https://github.com/wordpress-mobile/WordPress-iOS/pull/24165 .
     @objc var wordPressComRestApi: WordPressComRestApi? {
         get {
-            guard let api = objc_getAssociatedObject(self, &apiAssociatedKey) as? WordPressComRestApi else {
-                guard authToken.isEmpty else {
-                    let api = WordPressComRestApi.defaultApi(
-                        oAuthToken: authToken,
-                        userAgent: WPUserAgent.defaultUserAgent(),
-                        localeKey: WordPressComRestApi.LocaleKeyDefault
-                    )
-
-                    api.setInvalidTokenHandler { [weak self] in
-                        guard let self else { return }
-
-                        self.authToken = nil
-                        WordPressAuthenticationManager.showSigninForWPComFixingAuthToken()
-
-                        guard self.isDefaultWordPressComAccount else { return }
-
-                        // At the time of writing, there is an implicit assumption on what the object parameter value means.
-                        // For example, the WordPressAppDelegate.handleDefaultAccountChangedNotification(_:) subscriber inspects the object parameter to decide whether the notification was sent as a result of a login.
-                        // If the object is non-nil, then the method considers the source a login.
-                        //
-                        // The code path in which we are is that of an invalid token, and that's neither a login nor a logout, it's more appropriate to consider it a logout.
-                        // That's because if the token is invalid the app will soon received errors from the API and it's therefore better to force the user to login again.
-                        NotificationCenter.default.post(
-                            name: NSNotification.Name.WPAccountDefaultWordPressComAccountChanged,
-                            object: nil
-                        )
-                    }
-
-                    return api
-                }
-
+            if let api = objc_getAssociatedObject(self, &apiAssociatedKey) as? WordPressComRestApi {
+                return api
+            }
+            guard !authToken.isEmpty else {
                 DispatchQueue.main.async {
                     WordPressAuthenticationManager.showSigninForWPComFixingAuthToken()
                 }
                 return nil
             }
+            let api = makeWordPressComRestApi()
+            self.wordPressComRestApi = api
             return api
         }
         set(api) {
             objc_setAssociatedObject(self, &apiAssociatedKey, api, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
         }
     }
 
+    private func makeWordPressComRestApi() -> WordPressComRestApi {
+        let api = WordPressComRestApi.defaultApi(
```

Comment:
To test: 

- Set a breakpoint and verify that this gets called ~one time on app launch
- Verify that it gets nilled out on logout

Reply:
The setter does seem to be called constantly on `didDidTurnInfoFault` – odd.